src/jit/morph.cpp

```diff
@@ -6925,15 +6925,121 @@ void Compiler::fgMorphCallInlineHelper(GenTreeCall* call, InlineResult* result)
 #endif
 }
 
-/*****************************************************************************
- *
- * Performs checks to see if this tail call can be optimized as epilog+jmp.
- */
+//------------------------------------------------------------------------
+// fgCanFastTailCall: Check to see if this tail call can be optimized as epilog+jmp.
+//
+// Arguments:
+//    callee - The callee to check
+//
+// Return Value:
+//    Returns true or false based on whether the callee can be fastTailCalled
+//
+// Notes:
+//    This function is target specific and each target will make the fastTailCall
+//    decision differently. See the notes below.
+//
+//
+// Windows Amd64:
+//    A fastTailCall decision can be made whenever the amount of callee arguments
+//    is larger than or equal to the amount of caller arguments or we have four
+//    or less callee arguments.
```

Comment:
A suggested reformulation:
```
// A fast tail call can be made whenever the number of callee arguments
// is larger than or equal to the number of caller arguments, or we have four
// or fewer callee arguments. This is because, on Windows AMD64, each argument
// uses exactly one register or one 8-byte stack slot. Thus, we only need to count
// arguments, and not be concerned with the size of each incoming or outgoing
// argument.
```
Is my statement correct?